include/mbedtls/cipher.h

```diff
@@ -392,8 +400,8 @@ void mbedtls_cipher_free( mbedtls_cipher_context_t *ctx );
  *                      structure with the appropriate values. It also clears
  *                      the structure.
  *
- * \param ctx           The context to initialize. May not be NULL.
- * \param cipher_info   The cipher to use.
+ * \param ctx           The context to initialize. This must be initialized.
+ * \param cipher_info   The cipher to use. This may not be \c NULL.
```

Comment:
After discussing this with @mpg and @gilles-peskine-arm internally, this assumption should be removed. Instead, it should be documented behaviour that the function returns `#MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA` if you pass `NULL` here.